pghoard/restore.py

```diff
@@ -698,6 +708,9 @@ def fetch_all(self):
             except TimeoutError:
                 self.pending_jobs.clear()
                 self.last_progress_ts = time.monotonic()
+
+                # Increase the timeout and retry
+                self.max_stale_seconds *= 2
```

Comment:
Should it be configurable, and, by default, use the previous logic? Just to not break "backward compatibility," because someone might not need it and use the default pattern. But I might be wrong.

Reply:
All this does is increase the timeout when we get a timeout error so that the next time the loop executes it has longer to run. We saw on databases with large backups it was timing out and could not complete the download unless we bumped this value.